plugins/promela/hu.bme.mit.gamma.xsts.promela.transformation/src/hu/bme/mit/gamma/xsts/promela/transformation/util/ParallelActionHandler.xtend

```diff
@@ -0,0 +1,89 @@
+package hu.bme.mit.gamma.xsts.promela.transformation.util
+
+import hu.bme.mit.gamma.expression.model.Declaration
+import hu.bme.mit.gamma.expression.util.ExpressionUtil
+import hu.bme.mit.gamma.util.GammaEcoreUtil
+import hu.bme.mit.gamma.xsts.model.Action
+import hu.bme.mit.gamma.xsts.model.CompositeAction
+import hu.bme.mit.gamma.xsts.model.IfAction
+import hu.bme.mit.gamma.xsts.model.LoopAction
+import hu.bme.mit.gamma.xsts.model.MultiaryAction
+import hu.bme.mit.gamma.xsts.model.ParallelAction
+import hu.bme.mit.gamma.xsts.model.VariableDeclarationAction
+import hu.bme.mit.gamma.xsts.promela.transformation.serializer.DeclarationSerializer
+import java.util.List
+import java.util.Map
+
+class ParallelActionHandler {
+	// Singleton
+	public static final ParallelActionHandler INSTANCE = new ParallelActionHandler
```

Comment:
There is clearly an internal state for the instances of this class, so do not make it singleton